fluent/src/langneg.js

```diff
@@ -0,0 +1,246 @@
+
+/**
+ * Likely subtags map from a locale in minimized form to its maximized form.
+ * It's useful for non-intuitive matches like 'en' -> 'en-US' or marking
+ * what's the languge default region and script.
+ */
+const likelySubtags = {
+  'en': 'en-Latn-US',
+  'sr': 'sr-Cyrl-RS',
+  'sr-RU': 'sr-Latn-RU',
+  'az': 'az-Latn-AZ',
+  'az-IR': 'az-Arab-IR',
+  'zh': 'zh-Hans-CN',
+  'zh-GB': 'zh-Hant-GB',
+  'zh-US': 'zh-Hant-US',
+};
+
+/**
+ * Attempts to build a most likely maximum version of the locale id based
+ * on it's minimal version.
+ *
+ * If possible it uses the lookup table above. If this one is missing,
+ * it only attempts to place the language code in the region position.
+ */
+function addLikelySubtags(locale) {
+  if (likelySubtags.hasOwnProperty(locale)) {
+    return likelySubtags[locale];
+  }
+  const loc = parseLocale(locale, true);
+  if (loc[2] === '*') {
+    loc[2] = loc[0].toUpperCase();
+    return loc.join('-');
+  }
```

Comment:
So, here's the thing. LikelySubtags is a 50kb uncompressed JSON table [0].
If we have it (Gecko does), or in case where the user will want to pay the price (give me fluent with likelySubtags), then we don't need a dummy code like that.

But when the user doesn't want to pay the price, and 50kb is a pretty hefty price to pay on the client side, lack of this dummy code will make us not be able to catch any kind of 'ab-AB' extrapolation from 'ab'. 
Since it's unlikely that anyone will provide 'ca-CA' or 'lij-LIJ' in available locales, there's no consequence of testing against it. But in the simplest form, they'll at least match all 'fr' to 'fr-FR', and 'it' to 'it-IT' and 'de' to 'de-DE' etc.

My idea was to allow for three "modes":
* Full CLDR likelySubtags (50kb price for excellent matching)
* Limited CLDR likelySubtags (say, 10-20 most common ones) + dummy
* Just dummy

I can see us curating this limited CLDR likelySubtags just to catch the most common ones like 'en' into 'en-US', and 'fr' into 'fr-FR' etc.

I believe that if we remove the dummy all together, we'll make the algorithm not work for most common cases and basically require the likelySubtags.

What do you think?

[0] https://github.com/unicode-cldr/cldr-core/blob/master/supplemental/likelySubtags.json